src/hmpps/assets/javascript/datepicker.js

```diff
@@ -0,0 +1,608 @@
+            <button class="js-datepicker-prev-year" aria-label="previous year" data-button="button-datepicker-prevyear">
+                <span class="govuk-visually-hidden">Previous year</span>
+                <svg focusable="false" class="hmpps-datepicker-icon" aria-hidden="true" role="img"><use href="/assets/images/icons.stack.svg#double_chevron_left"></use></svg>
+            </button>
+
+            <button class="js-datepicker-prev-month" aria-label="previous month" data-button="button-datepicker-prevmonth">
+                <span class="govuk-visually-hidden">Previous month</span>
+                <svg focusable="false" class="hmpps-datepicker-icon" aria-hidden="true" role="img"><use href="/assets/images/icons.stack.svg#chevron_left"></use></svg>
+            </button>
+        </div>
+
+        <h2 id="${titleId}" class="hmpps-datepicker__dialog__title  js-datepicker-month-year" aria-live="polite">June 2020</h2>
+
+        <div class="hmpps-datepicker__dialog__navbuttons">
+            <button class="js-datepicker-next-month" aria-label="next month" data-button="button-datepicker-nextmonth">
+                <span class="govuk-visually-hidden">Next month</span>
+                <svg focusable="false" class="hmpps-datepicker-icon" aria-hidden="true" role="img"><use href="/assets/images/icons.stack.svg#chevron_right"></use></svg>
+            </button>
+
+            <button class="js-datepicker-next-year" aria-label="next year" data-button="button-datepicker-nextyear">
+                <span class="govuk-visually-hidden">Next year</span>
+                <svg focusable="false" class="hmpps-datepicker-icon" aria-hidden="true" role="img"><use href="/assets/images/icons.stack.svg#double_chevron_right"></use></svg>
+            </button>
+        </div>
+      </div>
+
+      <table class="hmpps-datepicker__dialog__table  js-datepicker-grid" role="grid" aria-labelledby="${titleId}">
+      <caption class="hmpps-datepicker__dialog__table-caption">You can use the arrow keys to select a date</caption>
+      <thead>
+          <tr>
+          <th scope="col" abbr="Monday">Mo</th>
+          <th scope="col" abbr="Tuesday">Tu</th>
+          <th scope="col" abbr="Wednesday">We</th>
+          <th scope="col" abbr="Thursday">Th</th>
+          <th scope="col" abbr="Friday">Fr</th>
+          <th scope="col" abbr="Saturday">Sa</th>
+          <th scope="col" abbr="Sunday">Su</th>
+          </tr>
+      </thead>
+
+      <tbody></tbody>
+      </table>
+
+      <div class="hmpps-datepicker__dialog__buttongroup">
+      <button type="button" class="govuk-button js-datepicker-ok" value="ok" data-button="button-datepicker-ok">Select</button>
+      <button type="button" class="govuk-button govuk-button--secondary js-datepicker-cancel" value="cancel" data-button="button-datepicker-cancel">Cancel</button>
+      </div>`
+}
+
+Datepicker.prototype.leadingZeroes = function (value, length = 2) {
+  let ret = value.toString()
```

Comment:
is this method an implementation of https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/padStart ?

Reply:
good spot, it may be!  I pretty much lifted all this from the DPS code as-is. 

There is some ongoing work with this because I believe other projects have identified bugs but just fixed it in their own local repo, so I will be bringing their fixes in too. I'll add into the new ticket that this code may be able to be replaced with padStart